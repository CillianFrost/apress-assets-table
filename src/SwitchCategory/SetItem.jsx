import React from 'react';
import PropTypes from 'prop-types';
import { block } from '../utils';
import Checkbox from '../Checkbox/Checkbox';

const bm = block('e-switch-cat-menu');

const SetItem = (props) => {
  const {
    onSelect,
    mix,
    checked,
    title,
    children,
  } = props;

  return (
    <div
      onClick={onSelect}
      className={bm('type').mix(mix)()}
      role="presentation"
    >
      <div className={bm('radio-set')()}>
        <Checkbox
          mix={bm('radio').mix('is-radio')()}
          checked={checked}
          onChange={() => {}}
        />
        <div>{title}</div>
      </div>
      <label
        className="e-label"
        htmlFor
      >
        <div className={bm('description')()}>
          {children}
        </div>
      </label>
    </div>
  );
};

SetItem.propTypes = {
  onSelect: PropTypes.func,
  mix: PropTypes.string,
  checked: PropTypes.bool,
  title: PropTypes.node,
};

export default SetItem;
