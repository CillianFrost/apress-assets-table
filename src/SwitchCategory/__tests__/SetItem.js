import React from 'react';
import {shallow} from 'enzyme';
import SetItem from '../SetItem';

describe('component SetItem', () => {
  const onSelect = jest.fn();
  const component = shallow(
    <SetItem
      onSelect={onSelect}
      title={'Заголовок'}
      checked={false}
    >
        description
      </SetItem>
    );

    it('Должен рендриться', () => {
      expect(component).toMatchSnapshot();
    });

    it('Должен обрабатывать onSelect', () => {
      component.find('.e-switch-cat-menu-type').simulate('click');
      expect(onSelect).toBeCalled();
    });
});
