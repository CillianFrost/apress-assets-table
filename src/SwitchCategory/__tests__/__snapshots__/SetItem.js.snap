// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component SetItem Должен рендриться 1`] = `
<div
  className="e-switch-cat-menu-type"
  onClick={[Function]}
>
  <div
    className="e-switch-cat-menu-radio-set"
  >
    <Checkbox
      checked={false}
      mix="e-switch-cat-menu-radio is-radio"
      onChange={[Function]}
    />
    <div>
      Заголовок
    </div>
  </div>
  <label
    className="e-label"
    htmlFor={true}
  >
    <div
      className="e-switch-cat-menu-description"
    >
      description
    </div>
  </label>
</div>
`;
